### DISCUSSION TITLE
Some super crazy ideas I have about this

### DISCUSSION THREAD

**Discussion by Isakku** [GitSquared/edex-ui#823] [Ideas]
Even though I searched for the "Ideas" tag I didn't find anything like this.

Ever since I was finishing my career, I had something like this project in mind. However, it was more rudimentary, more like making the terminal like a chat. Now tech seems to be advanced enough for the idea to be a full blown assistant besides of being possible  to explain the fundaments that could make it work in better terns:

**Perceived idea awesomeness**

1) Add voice recognition capabilities so this CLI can be activated by voice commands in a voice assistant fashion if desired.

2) Add speech emulation capabilities, so this terminal output results can be listened in spoken language if wanted.

3) Add GPT-3 similar AI capabilities so even if you don't remember the exact way a command is supposed to work, you can ask for what you want and the system will auto-complete or even compose the command just by asking for the details (and reading aloud or showing the man pages in another window if you ask about what a certain parameter in the command does) (Sadly Open AI was captured by Microsoft and the GPT-3 code was never released)
https://twitter.com/elonmusk/status/1309052632850468864

4) For the sake of freedom make sure it can all work offline. Perhaps by keeping all of the above working in minimal functionality or something like that. It should also keep working like a regular written input command line terminal with an  autocomplete function for commands as well. So for example if you type sudo apt-get the terminal suggests both update and upgrade and you simply choose one from the list to autocomplete

5) Does a Star Trek theme for this exists? If not then it probably should. I would really like to say aloud something like 

-"Computer! Go to directory /home/MyUsername/music/rock/" 
and then say
-"Computer! Open File jazzrock.m3u" 
and it will open it in the default media player.

Or
-"Computer! Go to folder y /home/MyUsername/Vídeos/"
and then 
-"Computer! Open File Composition.mkv On Screen!"

Notice that the Open File voice command is equivalent to double clicking a file: the default program that opens it should start for example VLC.
The On Screen command could be anything, I'm just making this up. 
EDIT
It could be a way to tell the command to start VLC and go into full screen perhaps but VLC lacks such CLI option.

Oh and how about composing new scripts? GPT-3 can already code, if this could have even very basic simmilar capabilities it could at least suggest bash templates for the user to make the scripting just like an IDE.

- [ ] Wow, you don't have this yet???
- [x] Would be super cool.
- [x] I'm not sure a lot of people would use it, but I'd love to have it.
- [x] Kind of a gadget, honestly, but why not?

**Perceived idea difficulty**

I know GPT-3 is extremely complex, but I think a less complex version of that NLP deep learning neural network to recognize commands and regular related words could be enough.

- [ ] Quit your job first.
- [ ] Would need some work, but nothing impossible.
- [ ] Shouldn't be too hard to make.
- [ ] Like, *I* could do it.
- [x] It should be possible, but it may require a whole non profit organization backing it up, otherwise it's impossible.

The most important part of this idea is that it should remain open source for the sake of humankind, otherwise it may end up just like Open AI and GitHub: absorbed by a big company like Microsoft. (hey this is sad.)

I know this is a huge dream (a pipedream maybe) and perhaps it will never be possible to make this. Still I wanted to make sure this wish is published. A non profit organization... A kickstarter campaign to create such a thing maybe? But for that to work we need the idea to be known and to become popular enough for people to fund it and to keep donating afterwards to maintain it.

**Comment by StoiaCode** [comment]
for 1. i reccomend VoiceAttack. Shouldnt be all too hard to at least get very basic commands to work

**Comment by wrac4242** [comment]
how would this translate over accents, i.e. scottish accents, thick accents

and how would it deal with ambiguity?

**Reply by antares667367** [reply]
i believe it could be done with some training, like with professional text dictation software.

**Reply by WalrusGumboot** [reply]
ELEHVAN
